File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        List<Integer> ans = new LinkedList<>();
        if(root==null)
            return ans;
        LinkedList<Node> st = new LinkedList<>();
        st.push(root);
        while(!st.isEmpty()) {
            Node top = st.pollLast();
            ans.addFirst(top.val);
            for(Node n:top.children)
                st.add(n);
        }
        return ans;
    }
}

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<int> postorder(Node* root) {
        if (root == NULL)
            return vector<int>();
        stack<Node*> st;
        st.push(root);
        list<int> ans;
        while(!st.empty()) {
            Node* top = st.top();
            st.pop();
            ans.push_front(top->val);
            for(auto it=top->children.begin();it!=top->children.end();it++) {
                st.push(*it);
            }
        }
        vector<int> finalAns;
        for(auto it=ans.begin();it!=ans.end();it++)
            finalAns.push_back(*it);
        return finalAns;
    }
};